cloudinit/sources/helpers/vultr.py

```diff
@@ -276,8 +276,10 @@ def generate_interface_additional_addresses(interface, netcfg):
 
 
 # Make required adjustments to the network configs provided
-def add_interface_names(interfaces):
-    for interface in interfaces:
+def add_interface_names(netcfg):
+    for interface in netcfg["config"]:
```

Comment:
This change implies that either:

1. This function has never been functional, or
2. There was a change in the instance metadata schema under `cloud_interfaces`. See https://github.com/canonical/cloud-init/blob/aacdff5b9298db2acbb5d10f0a42cb06d2c0a745/cloudinit/sources/DataSourceVultr.py#L73

Reply:
Its never been functional nor has it ever been used. This is our way of dropping in a fully configured, outside of interface names, cloud-init netcfg file in. Its not currently documented anywhere. As for unit tests, I will add that now.

Reply:
Thanks for the context and the unit test.